fiftyone/core/collections.py

```diff
@@ -1008,27 +1033,28 @@ def one(self, expr, exact=False):
             and multiple samples match the expression
 
         Returns:
-            a :class:`fiftyone.core.sample.SampleView`
+            a class:`fiftyone.core.sample.Sample` or
+            :class:`fiftyone.core.sample.SampleView`
         """
-        view = self.match(expr)
-        matches = iter(view)
+        limit = 2 if exact else 1
+        view = self.match(expr).limit(limit)
+        matches = iter(view._aggregate(detach_frames=True, detach_groups=True))
 
         try:
-            sample = next(matches)
+            d = next(matches)
         except StopIteration:
             raise ValueError("No samples match the given expression")
 
         if exact:
             try:
                 next(matches)
```

Comment:
Nice! Is there a way to check if the iterator still has more elements instead of calling next()?